deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+            additional_options = [[] for _ in range(desired_num_options - num_options)]
+            self.answers_indexed += additional_options
+        # Pad the number of tuples per option
+        for curr_answer in range(len(self.answers_indexed)):
+            self.answers_indexed[curr_answer] = self.pad_num_tuples(self.answers_indexed[curr_answer],
+                                                                    desired_num_question_tuples)
+        # Pad each of the tuples for slot length and words per slot
+        for answer_index in range(len(self.answers_indexed)):
+            self.answers_indexed[answer_index] = [self.pad_tuple(answer_tuple, desired_num_slots,
+                                                                 desired_question_slot_length)
+                                                  for answer_tuple in self.answers_indexed[answer_index]]
+        # Pad the number of background tuples
+        self.background_indexed = self.pad_num_tuples(self.background_indexed, desired_num_background_tuples)
+        # Pad each of the background tuples for number of slots and slot length
+        self.background_indexed = [self.pad_tuple(background_tuple, desired_num_slots,
+                                                  desired_background_slot_length)
+                                   for background_tuple in self.background_indexed]
+
+    # Helper method to pad a list of tuples to the desired number of tuples
+    @staticmethod
+    def pad_num_tuples(tuples_in, desired_length):
+        num_tuples = len(tuples_in)
+        # If there are more than desired, we remove them
+        if num_tuples > desired_length:
+            tuples_in = tuples_in[:desired_length]
+        elif num_tuples < desired_length:
+            additional_tuples = [[] for _ in range(desired_length - num_tuples)]
+            tuples_in += additional_tuples
+        return tuples_in
+
+    # Helper method to pad an individual tuple both to the desired number of slots as well as the slot length
+    def pad_tuple(self, tuple_in: List[List[int]], desired_num_slots, desired_slot_length):
+        num_slots = len(tuple_in)
+        if num_slots > desired_num_slots:
+            # We concatenate the indices from the slots at the end to make word_indices the right
+            # length.
+            num_extra_slots = num_slots - desired_num_slots
+            all_slots = tuple_in
+            tuple_in = all_slots[:desired_num_slots]
+            for i in range(num_extra_slots):
+                tuple_in[desired_num_slots - 1] += all_slots[desired_num_slots + i]
+        elif num_slots < desired_num_slots:
+            additional_slots = [[] for _ in range(desired_num_slots - num_slots)]
+            tuple_in += additional_slots
+        # Pad the slots to the desired length
+        max_lengths_to_use = {'word_sequence_length': desired_slot_length}
+        tuple_in = [self.pad_word_sequence(indices, max_lengths_to_use) for indices in tuple_in]
```

Comment:
would that be to pad the words to a set character length? if so, is there a super class method for that padding?